serde/src/ser/mod.rs

```diff
@@ -41,7 +41,7 @@ pub trait Serialize {
 ///////////////////////////////////////////////////////////////////////////////
 
 /// A trait that describes a type that can serialize a stream of values into the underlying format.
-pub trait Serializer {
+pub trait Serializer: Sized {
```

Comment:
Yes adding a `TaggedVisitor` is possible. In this case the `Sized` bound can be removed.